File: Upcoder/Khối 11/--/Xét điểm tuyển sinh/IF_Tuyensinh.py
def main():
    # Nhập điểm chuẩn
    diem_chuan = float(input())

    # Nhập điểm thi 3 môn
    diem_mons = list(map(float, input().split()))
    diem_mon1, diem_mon2, diem_mon3 = diem_mons

    # Nhập khu vực
    khu_vuc = input().strip()

    # Nhập đối tượng
    doi_tuong = int(input())

    # Tính điểm ưu tiên
    diem_uu_tien = 0
    if khu_vuc == 'A':
        diem_uu_tien += 2
    elif khu_vuc == 'B':
        diem_uu_tien += 1
    elif khu_vuc == 'C':
        diem_uu_tien += 0.5

    if doi_tuong == 1:
        diem_uu_tien += 2.5
    elif doi_tuong == 2:
        diem_uu_tien += 1.5
    elif doi_tuong == 3:
        diem_uu_tien += 1

    # Tính điểm tổng kết
    tong_diem = diem_mon1 + diem_mon2 + diem_mon3 + diem_uu_tien

    # Kiểm tra điều kiện đậu
    if tong_diem >= diem_chuan and diem_mon1 > 0 and diem_mon2 > 0 and diem_mon3 > 0:
        print("PASS", end="")
    else:
        print("FAIL", end="")

if __name__ == "__main__":
    main()

File: Upcoder/Khối 11/--/Xét điểm tuyển sinh/IF_Tuyensinh.cs
using System;

class Program
{
    static void Main()
    {
        // Nhập điểm chuẩn
        double diemChuan = double.Parse(Console.ReadLine());

        // Nhập điểm thi 3 môn
        string[] diemMon = Console.ReadLine().Split();
        double diemMon1 = double.Parse(diemMon[0]);
        double diemMon2 = double.Parse(diemMon[1]);
        double diemMon3 = double.Parse(diemMon[2]);

        // Nhập khu vực
        char khuVuc = char.Parse(Console.ReadLine());

        // Nhập đối tượng
        int doiTuong = int.Parse(Console.ReadLine());

        // Tính điểm ưu tiên
        double diemUuTien = 0;
        switch (khuVuc)
        {
            case 'A':
                diemUuTien += 2;
                break;
            case 'B':
                diemUuTien += 1;
                break;
            case 'C':
                diemUuTien += 0.5;
                break;
        }

        switch (doiTuong)
        {
            case 1:
                diemUuTien += 2.5;
                break;
            case 2:
                diemUuTien += 1.5;
                break;
            case 3:
                diemUuTien += 1;
                break;
        }

        // Tính điểm tổng kết
        double tongDiem = diemMon1 + diemMon2 + diemMon3 + diemUuTien;

        // Kiểm tra điều kiện đậu
        if (tongDiem >= diemChuan && diemMon1 > 0 && diemMon2 > 0 && diemMon3 > 0)
        {
            Console.WriteL("PASS");
        }
        else
        {
            Console.Write("FAIL");
        }
    }
}
